pubnub-kotlin-core/core-impl/src/main/kotlin/com/pubnub/internal/PNConfiguration.kt

```diff
@@ -0,0 +1,423 @@
+
+        /**
+         * If operating behind a misbehaving proxy, allow the client to shuffle the subdomains.
+         *
+         * Defaults to `false`.
+         */
+        var cacheBusting = false
+
+        /**
+         * When `true` the SDK doesn't send out the leave requests.
+         *
+         * Defaults to `false`.
+         */
+        var suppressLeaveEvents = false
+
+        /**
+         * When `true` the SDK will resend the last channel state that was set using [PubNub.setPresenceState]
+         * for the current [userId] with every automatic heartbeat (if [heartbeatInterval] is greater than 0)
+         * and initial subscribe connection (also after e.g. loss of network).
+         *
+         * Applies only when [enableEventEngine] is true.
+         *
+         * Defaults to `true`.
+         *
+         * Please note that `maintainPresenceState` doesn't apply to state that was set on channel groups.
+         * It is recommended to disable this option if you set state for channel groups using [PubNub.setPresenceState],
+         * otherwise that state may be overwritten by individual channel states.
+         */
+        var maintainPresenceState = true
+
+        /**
+         * Feature to subscribe with a custom filter expression.
+         */
+        var filterExpression: String = ""
+
+        /**
+         * Whether to include a [InternalPubNubClient.instanceId] with every request.
+         *
+         * Defaults to `false`.
+         */
+        var includeInstanceIdentifier = false
+
+        /**
+         * Whether to include a [InternalPubNubClient.requestId] with every request.
+         *
+         * Defaults to `true`.
+         */
+        var includeRequestIdentifier = true
+
+        /**
+         * Sets how many times to retry to reconnect before giving up.
+         * Must be used in combination with [reconnectionPolicy].
+         *
+         * The default value is `-1` which means unlimited retries.
+         */
+        @Deprecated(
+            level = DeprecationLevel.WARNING,
+            message = """Instead of reconnectionPolicy and maximumReconnectionRetries use retryConfiguration 
+            e.g. config.retryConfiguration = RetryConfiguration.Linear(delayInSec = 3, maxRetryNumber = 5) 
+            or config.retryConfiguration = RetryConfiguration.Exponential(minDelayInSec = 3, maxDelayInSec = 10, maxRetryNumber = 5)""",
+        )
+        var maximumReconnectionRetries = -1
+
+        /**
+         * @see [okhttp3.Dispatcher.setMaxRequestsPerHost]
+         */
+        var maximumConnections: Int? = null
+
+        /**
+         * Enable Google App Engine networking.
+         *
+         * Defaults to `false`.
+         */
+        var googleAppEngineNetworking = false
+
+        /**
+         * Whether to start a separate subscriber thread when creating the instance.
+         *
+         * Defaults to `true`.
+         */
+        var startSubscriberThread = true
```

Comment:
Do we need this? Maybe deprecation message can be added here?
If so it would be good to update Doc.

Reply:
removing this as it's not used with EE